optimum/exporters/openvino/model_patcher.py

```diff
@@ -3378,3 +3380,102 @@ def __exit__(self, exc_type, exc_value, traceback):
         super().__exit__(exc_type, exc_value, traceback)
         for block in self._model.model.layers:
             block.self_attn.forward = block.self_attn._orig_forward
+
+
+class Qwen2VLLanguageModelPatcher(DecoderModelPatcher):
+    def __init__(
+        self,
+        config: OnnxConfig,
+        model: Union[PreTrainedModel, TFPreTrainedModel],
+        model_kwargs: Dict[str, Any] = None,
+    ):
+        model.__orig_forward = model.forward
+
+        def forward_wrap(
+            self,
+            attention_mask,
+            position_ids=None,
+            past_key_values=None,
+            inputs_embeds=None,
+            input_ids=None,
+        ):
+            from transformers.cache_utils import DynamicCache
+
+            new_past_key_values = DynamicCache.from_legacy_cache(past_key_values)
```

Comment:
this can be directly included in `DecoderModelPatcher` instead of introducing `Qwen2VLLanguageModelPatcher` no ?

Reply:
I do not want to introduce these changes globally as expect that it will be supported in optimum itself and do not want spend more time for testing